### ISSUE TITLE
Add mentions of how to disable/enable SSR for different environments (ssr/prerender properties)

### ISSUE THREAD

**Issue by chriswoodie** [angular/angular#52884]
### Describe the problem that you experienced

Understanding how SSR works for v17+ is difficult. 

### Enter the URL of the topic with the problem

https://angular.dev/guide/ssr

### Describe what you were looking for in the documentation

I was trying to find how to disable SSR while serving an application during development.

### Describe the actions that led you to experience the problem

I removed old SSR configuration in my project and ran the setup SSR generator again, only to notice that SSR was now enabled by default (which has been mentioned elsewhere) but I did not think this would apply to development as it slows down reload times by A LOT.

### Describe what you want to experience that would fix the problem

Mention the 

          "ssr": false,
          "prerender": false,
          
configuration options. These were handed to me from a team member over on StackOverflow, but should really be part of the docs.

### Provide any additional information here in as much as detail as you can

```true
Would also be useful to mention why the old executors were dropped (such as serve-ssr) and what the new way of serving the application using SSR is and how it works in a bit more detail.
```

**Comment by alan-agius4** [comment]
SSR is not enabled by default. It is only enabled when opted in. When using the new application builder the performance implications of enabling SSR are minimal. 

Likely these options should be part of https://angular.dev/cli/commands which is currently not working properly.

**Comment by alan-agius4** [comment]
Update the options are available in https://angular.dev/cli/build, thus we could potentially link to these from the ssr / prerendering guide.

**Comment by imrangen** [comment]
This might help others - In Angular 17 ng serve is used to serve SSR application

Documentation: https://angular.dev/guide/ssr

**To verify that the application is server-side rendered, run it locally with ng serve. The initial HTML request should contain application content.**

**Comment by worthy7** [comment]
prerender input | ssr input | ssr routes input |   | expected:single /features page should   prerender? | expected:other pages should prerender? |   | Actual:single /features page actually   prerenders? | Actual:other pages actually prerender?
-- | -- | -- | -- | -- | -- | -- | -- | --
FALSE | FALSE | * |   | FALSE | FALSE |   | FALSE | FALSE
TRUE | FALSE | * |   | TRUE | TRUE |   | TRUE | TRUE
discoverRoutes:false, no routes file   specified | FALSE | * |   | FALSE | FALSE |   |**_TRUE_** | **_TRUE_**
discoverRoutes:false, routes emtpy | FALSE | * |   | FALSE | FALSE |   | **_TRUE_** | **_TRUE_**
discoverRoutes:false, routes file   '/features' | FALSE | * |   | TRUE | FALSE |   | TRUE | **_TRUE_**
FALSE | TRUE | ** client |   | FALSE | FALSE |   | FALSE | FALSE
FALSE | TRUE | ** prerender |   | TRUE | TRUE |   | TRUE | TRUE
FALSE | TRUE | features prerender, ** client |   | TRUE | FALSE |   | TRUE | FALSE
TRUE | TRUE | ** client |   | TRUE | TRUE |   | FALSE | FALSE



To summarize
If ssr is false, routes are ignored
prerender on anything except false, prerenders everything.. It shouldn’t
if ssr is true, prerender is ignored and server routes is used. It should be clear that prerender is not applicable in this case.

**Comment by worthy7** [comment]
My objective, is to get my homepage to prerender on `/` without having to set up a backend server.

if i do this with the `prerender` option, then `/dashboard` (which is behind a guard) will flash the homepage first and then show the dashboard.

Using ssr -> prerendering, it correctly only gives the prerendered homepage on `/` and not `/dashboard`, but requires a server.
